beacon-chain/execution/service.go

```diff
@@ -886,3 +899,33 @@ func (s *Service) migrateOldDepositTree(eth1DataInDB *ethpb.ETH1ChainData) error
 func (s *Service) removeStartupState() {
 	s.cfg.finalizedStateAtStartup = nil
 }
+
+func newBlobVerifierFromInitializer(ini *verification.Initializer) verification.NewBlobVerifier {
+	return func(b blocks.ROBlob, reqs []verification.Requirement) verification.BlobVerifier {
+		return ini.NewBlobVerifier(b, reqs)
+	}
+}
+
+type capabilityCache struct {
+	capabilities     []string
+	capabilitiesLock sync.RWMutex
+}
+
+func (c *capabilityCache) Save(cs []string) {
+	c.capabilitiesLock.Lock()
+	defer c.capabilitiesLock.Unlock()
+
+	c.capabilities = cs
+}
+
+func (c *capabilityCache) Has(capability string) bool {
+	c.capabilitiesLock.RLock()
+	defer c.capabilitiesLock.RUnlock()
+
+	for _, existingCapability := range c.capabilities {
+		if existingCapability == capability {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
Why not use a map? O(n) storage, O(1) lookup

```suggestion
type capabilityCache struct {
	capabilities     map[string]interface{}
	capabilitiesLock sync.RWMutex
}

func (c *capabilityCache) Save(cs []string) {
	c.capabilitiesLock.Lock()
	defer c.capabilitiesLock.Unlock()

	if c.capabilities == nil {
		c.capabilities = make(map[string]interface{})
	}
	c.capabilities[cs] = nil
}

func (c *capabilityCache) Has(capability string) bool {
	c.capabilitiesLock.RLock()
	defer c.capabilitiesLock.RUnlock()

	if c.capabilities == nil {
		return false
	}
	_, ok := c.capabilities[capability]
	return ok
}
```

Reply:
map is good too!